File: ModelConstants.java
package com.yourapp.ml;

import java.util.HashMap;
import java.util.Map;
import java.util.Arrays;
import java.util.List;

public class ModelConstants {
    
    // Target classes (output labels)
    public static final List<String> TARGET_CLASSES = Arrays.asList(
        "Insomnia",
        "Sleep Apnea",
        "nan"
    );
    
    // Gender mapping
    public static final Map<String, Integer> GENDER_MAPPING = new HashMap<String, Integer>() {{
        put("Female", 0);
        put("Male", 1);
    }};
    
    // Occupation mapping
    public static final Map<String, Integer> OCCUPATION_MAPPING = new HashMap<String, Integer>() {{
        put("Accountant", 0);
        put("Doctor", 1);
        put("Engineer", 2);
        put("Lawyer", 3);
        put("Manager", 4);
        put("Nurse", 5);
        put("Sales Representative", 6);
        put("Salesperson", 7);
        put("Scientist", 8);
        put("Software Engineer", 9);
        put("Teacher", 10);
    }};
    
    // BMI Category mapping
    public static final Map<String, Integer> BMI_MAPPING = new HashMap<String, Integer>() {{
        put("Normal", 0);
        put("Normal Weight", 1);
        put("Obese", 2);
        put("Overweight", 3);
    }};
    
    // Feature order (for reference)
    public static final List<String> FEATURE_ORDER = Arrays.asList(
        "Age",
        "Sleep Duration",
        "Quality of Sleep",
        "Physical Activity Level",
        "Stress Level",
        "Heart Rate",
        "Daily Steps",
        "Systolic_BP",
        "Diastolic_BP",
        "Gender_Encoded",
        "Occupation_Encoded",
        "BMI_Encoded"
    );
    
    // Helper methods
    public static int getGenderCode(String gender) {
        return GENDER_MAPPING.getOrDefault(gender, 0);
    }
    
    public static int getOccupationCode(String occupation) {
        return OCCUPATION_MAPPING.getOrDefault(occupation, 0);
    }
    
    public static int getBMICode(String bmiCategory) {
        return BMI_MAPPING.getOrDefault(bmiCategory, 0);
    }
    
    public static String getPredictedClass(int classIndex) {
        if (classIndex >= 0 && classIndex < TARGET_CLASSES.size()) {
            return TARGET_CLASSES.get(classIndex);
        }
        return "Unknown";
    }
}

File: ModelConstants.kt
package com.yourapp.ml

object ModelConstants {
    
    // Target classes (output labels)
    val TARGET_CLASSES = listOf(
        "Insomnia",
        "Sleep Apnea",
        "nan"
    )
    
    // Gender mapping
    val GENDER_MAPPING = mapOf(
        "Female" to 0,
        "Male" to 1
    )
    
    // Occupation mapping
    val OCCUPATION_MAPPING = mapOf(
        "Accountant" to 0,
        "Doctor" to 1,
        "Engineer" to 2,
        "Lawyer" to 3,
        "Manager" to 4,
        "Nurse" to 5,
        "Sales Representative" to 6,
        "Salesperson" to 7,
        "Scientist" to 8,
        "Software Engineer" to 9,
        "Teacher" to 10
    )
    
    // BMI Category mapping
    val BMI_MAPPING = mapOf(
        "Normal" to 0,
        "Normal Weight" to 1,
        "Obese" to 2,
        "Overweight" to 3
    )
    
    // Feature order (for reference)
    val FEATURE_ORDER = listOf(
        "Age",
        "Sleep Duration",
        "Quality of Sleep",
        "Physical Activity Level",
        "Stress Level",
        "Heart Rate",
        "Daily Steps",
        "Systolic_BP",
        "Diastolic_BP",
        "Gender_Encoded",
        "Occupation_Encoded",
        "BMI_Encoded"
    )
    
    // Helper functions
    fun getGenderCode(gender: String): Int = GENDER_MAPPING[gender] ?: 0
    
    fun getOccupationCode(occupation: String): Int = OCCUPATION_MAPPING[occupation] ?: 0
    
    fun getBMICode(bmiCategory: String): Int = BMI_MAPPING[bmiCategory] ?: 0
    
    fun getPredictedClass(classIndex: Int): String {
        return if (classIndex in TARGET_CLASSES.indices) {
            TARGET_CLASSES[classIndex]
        } else {
            "Unknown"
        }
    }
}
